src/nat/builder/per_user_workflow_builder.py

```
@@ -0,0 +1,745 @@
+                                                                component_type: str = "function"
+                                                                ) -> list[FunctionMiddleware]:
+        """
+        Resolve middleware names to FunctionMiddleware instances from the shared builder.
+        """
+        middleware_instances: list[FunctionMiddleware] = []
+        for middleware_name in middleware_names:
+            middleware_obj = await self._shared_builder.get_middleware(middleware_name)
+            if not isinstance(middleware_obj, FunctionMiddleware):
+                raise TypeError(f"Middleware `{middleware_name}` is not a FunctionMiddleware and cannot be used"
+                                f"with {component_type}s. "
+                                f"Only FunctionMiddleware types support function-specific wrapping.")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)